MainWindow.cpp

```diff
@@ -316,8 +330,25 @@ void MainWindow::openProject(std::unique_ptr<buffers::Project> openedProject) {
   treeModel.reset(new TreeModel(project->mutable_game()->mutable_root(), resourceMap.get(), nullptr));
 
   ui->treeView->setModel(treeModel.get());
+
   treeModel->connect(treeModel.get(), &TreeModel::ResourceRenamed, resourceMap.get(),
                      &ResourceModelMap::ResourceRenamed);
+
+  nodeResource.clear();
+  const QTreeView *treeUi = ui->treeView;
+  treeModel.get();
+  QModelIndex node = treeUi->indexAt(QPoint());
+  QVariant nodeString;
+  int index;
+  while (node.isValid()) {

```

Comment:
These are all things that you should be defensively programming and logging using `R_EXPECT` or `R_ASSESS`.